Create ArmyLegionToken.cs

Mercule opened the pull request:
Farm legion tokens with 3 methods: dreadrock, legion arena, and paragon shogun pet

Mercule committed:
Create ArmyLegionToken.cs

Farm legion tokens with 3 methods: dreadrock, legion arena, and paragon shogun pet

Mercule committed:
Merge branch 'BrenoHenrike:Skua' into Skua

Mercule committed:
Merge branch 'BrenoHenrike:Skua' into Skua

Mercule committed:
Merge branch 'BrenoHenrike:Skua' into Skua

Mercule committed:
Updated to new template

Mercule committed:
Merge branch 'BrenoHenrike:Skua' into Skua

Mercule committed:
Merge branch 'BrenoHenrike:Skua' into Skua

Mercule committed:
Fixed typo

LordExelot commented on Prototypes/ArmyLegionToken.cs:
```diff
@@ -0,0 +1,130 @@
+        Core.SetOptions(false);
+    }
+
+    public string[] Loot = { "Legion Token" };
```
It's literally one item, nothing is gained from putting this inside a public string[], just write Legion Token where applicable

LordExelot commented on Prototypes/ArmyLegionToken.cs:
```diff
@@ -0,0 +1,130 @@
+        Core.AddDrop(Loot);
+        if (Method.ToString() == "Dreadrock")
+        {
+            Core.Join("dreadrock");
```
Why not just move the join to inside the function

LordExelot commented on Prototypes/ArmyLegionToken.cs:
```diff
@@ -0,0 +1,130 @@
+        }
+        else if (Method.ToString() == "Shogun_Paragon_Pet")
+        {
+            Core.Join("fotia");
```
Why not just move the join to inside the function

LordExelot commented on Prototypes/ArmyLegionToken.cs:
```diff
@@ -0,0 +1,130 @@
+        }
+        else
+        {
+            Core.Join("legionarena");
```
Why not just move the join to inside the function

LordExelot commented on Prototypes/ArmyLegionToken.cs:
```diff
@@ -0,0 +1,130 @@
+        void dreadrock()
+        {
+        Core.RegisterQuests(4850);
+        if (string.IsNullOrEmpty(Bot.Config.Get<string>("player5").Trim()) && string.IsNullOrEmpty(Bot.Config.Get<string>("player6").Trim()))
```
As it is right now, it'll only use this one if at 4 players or less, the same could be achieved by checking only player 5

LordExelot commented on Prototypes/ArmyLegionToken.cs:
```diff
@@ -0,0 +1,130 @@
+
+        void legionarena()
+        {
+        Core.AddDrop("Legion Token", "Bone Sigil");
```
You already addDrop'd Legion Tokens

LordExelot commented on Prototypes/ArmyLegionToken.cs:
```diff
@@ -0,0 +1,130 @@
+        if (string.IsNullOrEmpty(Bot.Config.Get<string>("player5").Trim()) && string.IsNullOrEmpty(Bot.Config.Get<string>("player6").Trim()))
+            Army.AggroMonCells("Boss");
+        Army.AggroMonStart("legionarena");
+        Army.DivideOnCells("Boss");
```
If everyone is gonna go to a single cell, dont use divide, just use Core.Jump

LordExelot commented on Prototypes/ArmyTaintedGem.cs:
```diff
@@ -1,26 +1,35 @@
-        new Option<string>("player2", "Account #2", "Name of one of your accounts.", ""),
-        new Option<string>("player3", "Account #3", "Name of one of your accounts.", ""),
-        new Option<string>("player4", "Account #4", "Name of one of your accounts.", ""),
-        new Option<int>("PacketDelay", "Delay for Packet Spam", "Sets the delay for the Packet Spam \n" +
```
Missing packet delay option

LordExelot commented on Prototypes/ArmyTaintedGem.cs:
```diff
@@ -1,26 +1,35 @@
+        sArmy.player2,
+        sArmy.player3,
+        sArmy.player4,
+        sArmy.player5,
```
The original bot only has 4 players, are you sure this map can hold more?

LordExelot commented on Prototypes/ArmyTaintedGem.cs:
```diff
@@ -60,62 +69,39 @@ public void TaintedGem(int quant = 1000)
+        Army.AggroMonStart("boxes");
+        Army.DivideOnCells("Fort2", "Closet", "Fort1", "Boss");
+
         while (!Bot.ShouldExit && (!Core.CheckInventory("Cubes", 500)))
```
Not a problem made by you, but there should be a mechanism that causes them to wait when going to the next map, otherwise they might desync.
PR is still valid regardless

LordExelot commented on Prototypes/ArmyTaintedGem.cs:
```diff
@@ -60,62 +69,39 @@ public void TaintedGem(int quant = 1000)
+        if (string.IsNullOrEmpty(Bot.Config.Get<string>("player5").Trim()) && string.IsNullOrEmpty(Bot.Config.Get<string>("player6").Trim()))
+        Army.AggroMonCells("War");
+        Army.AggroMonStart("mountfrost");
+        Army.DivideOnCells("War");
```
Once more, if everyone is going to the same cell, use Core.Jump. Divide is way heavier and its just unnecessary

LordExelot reviewed (commented):
I left you some comments, please review and make the necessary changes. The one regarding the map desync is optional, as that one is tough

Mercule committed:
Update ArmyLegionToken.cs

Mercule committed:
Update ArmyTaintedGem.cs

The reason why I removed the waiting player function, because this boxes map have different drop rates on some cells and also the the monster hp is very low here, so sometimes the armies may miss some kills and cause it desync already. Also it only took few second on mountfrost map, so I think it would be not a problem. Already tried to run this script with player wait function and the farm speed is much slower than without player wait function.

And boxes map have max 6 player, previously I only add 4 players because I can only run 4 skua at same times. just added it for anyone who can run 6 clients.
